CHANGELOG.md

```diff
@@ -5,6 +5,34 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased] - ReleaseDate
+
+## [0.10.0] - December 12, 2023
+
+### Note
+
+`annotate-snippets-rs` is in the middle of a rewrite to make it more flexible and easier to use. 
+This release is a preview of the new API, and is not yet stable. 
+The API is subject to change, and the documentation is incomplete. 
+Please report any issues you find!
+
+### Added
+
+- `Renderer` is now used for displaying a `Snippet` [#67](https://github.com/rust-lang/annotate-snippets-rs/pull/67/commits/9076cbf66336e5137b47dc7a52df2999b6c82598)
+  - `Renderer` also controls the color scheme and formatting of the snippet
+
+### Breaking Changes
+
+- Changed the edition to `2021` [#61](https://github.com/rust-lang/annotate-snippets-rs/pull/61)
+- Set the minimum supported Rust version to `1.70.0` [#61](https://github.com/rust-lang/annotate-snippets-rs/pull/61)
```

Comment:
Not a breaking change (though a noteworthy, I tend to put these in a "compatibility" section)